src/org/rascalmpl/parser/gtd/preprocessing/ExpectBuilder.java

```diff
@@ -26,13 +28,15 @@
  */
 @SuppressWarnings("cast")
 public class ExpectBuilder<P>{
+	private final IntegerKeyedHashMap<IntegerList> dontNest;
 	private final IntegerMap resultStoreMappings;
 	
 	private final SortedIntegerObjectList<DoubleArrayList<P, AbstractStackNode<P>[]>> alternatives;
 	
-	public ExpectBuilder(IntegerMap resultStoreMappings){
+	public ExpectBuilder(IntegerKeyedHashMap<IntegerList> dontNest, IntegerMap resultStoreMappings){
```

Comment:
Propose to include the old constructor as well `@deprecated`, such that we can bootstrap without too many issues.

Reply:
Adding a constructor with the old signature, which uses an empty `dontNest` map, would make the change backwards compatible and should make everything work as before.
I considered this myself, but ultimately decided against it, since it (naturally) also retains the bug. Making this a breaking change forces people to regenerate any existing parsers and prevents them from continuing to use a deprecated/buggy version.
But feel free to add one if this is preferred.